### PR TITLE
Updated Mailinator to support Private Domain

### PR DESCRIPTION
**Pull request by mattrobbins**
Along with updating the gem to support the Private Domain I also updated the gem to handle the updated responses, added tests for the private domain upgrade, cleaned up a bunch of Rubocop errors and updated the documentation. This is a major change to I updated the version to 1.0.0.

Let me know if you approve or what you would like to see different.
Thanks!

### CODE REVIEW

**Comment by mattrobbins** [comment]
@ainformatico Can you take a look at this PR?

**Comment by ainformatico** [comment]
Hey @mattrobbins I've been busy these days, I'm really sorry! 

I see the CI build is red, can you maybe please check why?

I will take a look to this PR before the end of this week.

**Comment by ainformatico** [comment]
Hi again @mattrobbins! thanks a lot for your PR, I've taken a look and I also think that it makes sense to release `v1` and include some improvements such:

* Remove deprecated Ruby versions support
* Fix Rubocop issues
* Update gems
* Add the `private_domain` option
* General coding improvements

I have created https://github.com/ainformatico/mailinator/pull/14 and pointed your PR to that PR where I got some of your changes and also enabled some Code Analysis tools.
When I changed the destination of your branch I ran into some conflicts and had to solve them hence the new commit in your PR.

I would suggest we keep this PR _only_ with the changed needed for the `private_domain` option and leave the https://github.com/ainformatico/mailinator/pull/14 for the general changes of `v1`. What do you say?

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Lint/DuplicateMethods:
```
Comment (ainformatico) [inline .rubocop.yml]: I would say we can go with the _default_ community options, so no need for this file.

File: lib/mailinator/models/email.rb [hunk 2]
```diff
@@ -7,31 +7,21 @@ module Models
...
-          id: @data['data']['id'],
+          from_full: @data['data']['fromfull'],
+          date: DateTime.parse(@data['data']['headers']['date']),
+          received: @data['data']['headers']['received'],
+          from: @data['data']['headers']['from'],
+          to: @data['data']['headers']['to'],
           subject: @data['data']['subject'],
+          request_id: @data['data']['requestId'],
           body: @data['data']['parts'].first['body'],
-          body_html: retrieve_body_html,
```
Comment (ainformatico) [inline lib/mailinator/models/email.rb:13]: Any special reason for removing `f17d5b837f674eada25fe9e3e85cf262`?

File: spec/lib/mailinator/email_spec.rb [hunk 3]
```diff
@@ -10,24 +10,20 @@
...
+    expect(message.request_id).to eq('638363')
+    expect(message.body).to include('<p>This is a body</p>')
+    expect(message.body_base_64).to include('This is a BASE64 body JVBERi0xLjUKJe')
+    expect(message.orig_from).to eq('Sender <sender@example.com>')
+    expect(message.id).to eq('1419696967-44152505-recipient')
+    expect(message.time).to eq(1_526_072_140_000)
+    expect(message.seconds_ago).to eq(15_258)
   end
 
   it 'should return nil for #body_html when no html body' do
```
Comment (ainformatico) [inline spec/lib/mailinator/email_spec.rb]: If `body_html` needs to be removed then this test probably too.

File: spec/support/fake_web.rb [hunk 4]
```diff
@@ -6,8 +6,8 @@
 
 def fixture_file(filename)
   return '' if filename == ''
-
   file_path = File.join(__dir__, '../', 'fixtures/', filename)
+
```
Comment (ainformatico) [inline spec/support/fake_web.rb]: Oops, my bad in the merge.

File: README.md [hunk 5]
```diff
@@ -88,6 +91,13 @@ To delete an `Email` you need:
 inbox.messages.first.delete
 ```
 
+Access to a private inbox
+```ruby
+inbox = Mailinator::Inbox.get('inbox-abcd1234', private_domain: true)
+```
+
+The 'download' and 'delete' work the same with private domain.
```
Comment (ainformatico) [inline README.md:99]: I think it's worth to also add a section where it's explained that it is possible to pass any custom option, not only `private_domain`.

**Review by ainformatico — CHANGES_REQUESTED** [review]
@mattrobbins I just left some minor comments, please let me know if you need some help on applying them, for now I will focus on making the other changes needed for [v1](https://github.com/ainformatico/mailinator/pull/14)

**Comment by mattrobbins** [comment]
@ainformatico That sounds good to me. Sorry I was out on vacation all week last week and still getting caught up, which is why I haven't responded sooner. Do you want me to do another PR with just the private_domain updates or just update this one to have only the PR updates?

File: lib/mailinator/models/email.rb (hunk 2 above)
Comment (mattrobbins) [reply lib/mailinator/models/email.rb:13]: It appears the JSON has been updated to no longer have separate body and body_html elements. just the 'body'

File: README.md (hunk 5 above)
Comment (mattrobbins) [reply README.md:99]: Ok

File: .rubocop.yml (hunk 1 above)
Comment (mattrobbins) [reply .rubocop.yml]: Makes sense to me.

**Comment by ainformatico** [comment]
@mattrobbins I started to copy your changes over to https://github.com/ainformatico/mailinator/pull/14 and wanted to validate the new JSON provided by the API so I asked the Mailinator team for a trial token and still waiting for them to reply back. If you have a token already I would say let's use yours to get those fixtures updated (they already look updated in this very PR).
I would say we can just update this PR to have only the changes needed for the `private_domain` feature and tweak the couple of comments left.

**Comment by mattrobbins** [comment]
Ok Makes sense. I can update it to only have the private_domain feature. I do have a token as we are a paid customer. I won't be able to send you the token but I can send you the JSON's? One without the private_domain and one with. Will that work?

**Comment by ainformatico** [comment]
I just need the token so we can update all the fixtures, so if you can take care of that then perfect! Just remove the old fixtures and create the new ones. Don't forget to change the token from the fixtures as it is recorded in the url.

**Comment by mattrobbins** [comment]
@ainformatico I have updated the PR and merged the changes from you and resolved any conflicts. Looks like we are good to go. Let me know if I missed anything. Thanks!

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,6 @@
 /.bundle/
 /.yardoc
+/Gemfile.lock
```
Comment (ainformatico) [inline .gitignore]: This change should not be needed as it's already on my branch: https://github.com/ainformatico/mailinator/blob/v1-release/.gitignore#L3

File: README.md [hunk 7]
```diff
@@ -1,6 +1,5 @@
 [![Build Status](https://travis-ci.org/ainformatico/mailinator.svg)](https://travis-ci.org/ainformatico/mailinator)
 [![Code Climate](https://codeclimate.com/github/ainformatico/mailinator/badges/gpa.svg)](https://codeclimate.com/github/ainformatico/mailinator)
-[![Coverage Status](https://coveralls.io/repos/github/ainformatico/mailinator/badge.svg?branch=master)](https://coveralls.io/github/ainformatico/mailinator?branch=master)
```
Comment (ainformatico) [inline README.md]: This should not be removed as it has been added in the `v1` branch.

File: Rakefile [hunk 8]
```diff
@@ -7,4 +7,4 @@ RSpec::Core::RakeTask.new(:spec) do |task|
   task.rspec_opts = ['--color', '--format', 'documentation']
 end
 
-task default: :spec
```
Comment (ainformatico) [inline Rakefile]: I think this should be the good one.

File: lib/mailinator/api.rb [hunk 9]
```diff
@@ -4,6 +4,7 @@
 require 'json'
 
 module Mailinator
+
```
Comment (ainformatico) [inline lib/mailinator/api.rb]: Not really needed.

File: mailinator.gemspec [hunk 10]
```diff
@@ -9,7 +9,7 @@ Gem::Specification.new do |spec|
   spec.version       = Mailinator::VERSION
   spec.authors       = ['Alejandro Dev.']
   spec.email         = ['aeinformatico@gmail.com']
-  spec.summary       = 'Mailinator REST API wrapper'
+  spec.summary       = %q{Mailinator REST API wrapper}
```
Comment (ainformatico) [inline mailinator.gemspec]: Should not be changed, I think it comes from the old Rubocop config you had.

File: spec/lib/mailinator/email_spec.rb [hunk 11]
```diff
@@ -52,6 +50,7 @@
     let(:response) { subject.delete('abcd1234') }
 
     it 'deletes an email' do
+      response = Mailinator::Email.delete('abcd1234')
```
Comment (ainformatico) [inline spec/lib/mailinator/email_spec.rb]: This should be covered by the `let(:response) { subject.delete('abcd1234') }` on the lines above.

File: spec/lib/mailinator/model_spec.rb [hunk 12]
```diff
@@ -3,7 +3,8 @@
 require 'spec_helper'
 
 describe Mailinator::Models::Base do
-  let(:data) { { custom: :custom } }
+
+  let(:data) { {custom: :custom} }
```
Comment (ainformatico) [inline spec/lib/mailinator/model_spec.rb]: Not really necessary change IMHO.

File: spec/support/fake_web.rb [hunk 13]
```diff
@@ -4,8 +4,8 @@
...
```
Comment (ainformatico) [inline spec/support/fake_web.rb]: These lines should not be in this PR.

**Review by ainformatico — CHANGES_REQUESTED** [review]
Just left some _cosmetic_ changes so we can ensure we don't have conflicts or repeated lines between branches.
I would say after these comments are addressed you can just go ahead and _squash_ everything into a shorter list of commits that make sense or just a single commit. In case you prefer we can do it when merging by using Github's built-in button for merging with _rebasing_.

**Comment by ainformatico** [comment]
I think you should also merge my branch to ensure you have the latest changes.

**Comment by mattrobbins** [comment]
@ainformatico I made the changes requested. I appreciate the help you have given me. I think it would be easiest to do the squash with the Github button. Let me know if any other changes are needed. Thanks I am looking forward to this being complete.

**Pull request merged**